core/components/atoms/text-input/text-input.js

```diff
@@ -9,9 +9,9 @@ const TextInput = ({ defaultValue, ...props }) => {
   if (props.masked) {
     const length = defaultValue ? defaultValue.length : 8
     const maskedValue = new Array(length).join('•')
-    return <TextInput.Element {...props} placeholder={maskedValue} readOnly />
+    return <TextInput.Element type="text" {...props} placeholder={maskedValue} readOnly />
```

Comment:
This works perfectly 💯 

`<TextInput type="number"/>` renders a number input

Buuuut, do you think this might confuse folks, it looks like it's hard coded?

Suggestion: Add `type` in propTypes with `'text'` in `defaultProps`